### ISSUE TITLE
fastGlob('\\!dir/*') finds nothing even though a directory named !dir exists

### ISSUE THREAD

**Issue by thorn0** [mrmlnc/fast-glob#261]
### Environment

  * OS Version: WSL (10.0.18363.657) Ubuntu 18.04.1 LTS
  * Node.js Version: v10.15.0

### Actual behavior

Nothing found.

### Expected behavior

Files found.

### Steps to reproduce

  1. Create a non-empty dir named `!dir`
  2. Run the code sample below

### Code sample

```js
console.log(require("fast-glob").sync("\\!dir/*"))
```

**Comment by thorn0** [comment]
Workaround: `require("fast-glob").sync("@(!)dir1/*")`

**Comment by mrmlnc** [comment]
Looks like the problem around the `glob-parent` package. The first part of the pattern is defined as the base directory.

```
{ pattern: '\\!dir/*', base: '\\!dir' }
```

**Comment by anthonytranDev** [comment]
@thorn0 

Hey are you still working on this ticket. Was wondering whether I can help out?

Wanted to ask first - don't want to step over your toes

**Comment by thorn0** [comment]
I'm not working on it. It was just a bug report.

**Comment by mrmlnc** [comment]
I created a pull request to fix this problem in the `glob-parent` package (https://github.com/gulpjs/glob-parent/pull/26).

@thorn0 & @anthonytranDev, please tell me in which case you are use `!` in the directory name?

**Comment by mrmlnc** [comment]
Fixed by [`glob-parent@5.1.1`](https://github.com/gulpjs/glob-parent/commit/5b09270f53f956d11c36bff6bbb560fb53a13596).

In the simplest case, run this command to get the latest versions of dependencies:

```
npm i fast-glob
```
